### ISSUE TITLE
Suspect regression in upgrade from 2.10.9 to 2.11.0

### ISSUE THREAD

**Issue by bjornharrtell** [rollup/rollup#3610]
When upgrading from rollup 2.10.9 to 2.11.0 I get an unexpected build failure in my project that I've yet to be able to explain.

- Rollup Version: 2.10.9 / 2.11.0
- Operating System (or Browser): Ubuntu 20.04
- Node Version (if applicable): v12.17.0
- Link to reproduction (IMPORTANT, read below): https://github.com/bjornharrtell/flatgeobuf/pull/61

### Actual Behavior

Unexpected failure to build. Error message is:

```
Error: Entry module cannot be external (lib/generic.js).
    at error (/home/circleci/project/node_modules/rollup/dist/shared/rollup.js:217:30)
    at ModuleLoader.loadEntryModule (/home/circleci/project/node_modules/rollup/dist/shared/rollup.js:17552:20)
    at async Promise.all (index 0)
    at async Promise.all (index 0)
```

**Comment by lukastaegert** [comment]
It appears this is actually a bug in `@rollup/plugin-node-resolve`, more specifically in the `resolveOnly` option, that was uncovered by the new option normalization in v2.11.0. The bug is that the corresponding code is not capable of treating inputs that are arrays correctly. The result is that in your situation, an input of `lib/generic.js` is wrongly interpreted as the sub-path `generic.js` of a node_modules library `lib` which of course is considered external due to the `resolveOnly` option.

You can work around this easily for now in several ways:
- Add `"lib"` to the array in `resolveOnly`. This is like "working with the bug" in that you pretend that if it thinks it belongs to the `lib` package, resolve it.
- Make all inputs relative by prefixing them with `./`, e.g. `./lib/generic.js`. This will trigger a different code path that is not buggy.

**Comment by lukastaegert** [comment]
Fix at rollup/plugins#430

**Comment by bjornharrtell** [comment]
Great stuff. Thanks!
